src/js/health-records/containers/UpdatePage.jsx

```diff
@@ -21,28 +21,31 @@ export class UpdatePage extends React.Component {
   componentDidMount() {
     this.props.checkRefreshStatus();
     this.pollRefresh();
+    this.refreshTimeout = setTimeout(this.submitAndDownload.bind(this), 600000);
   }
 
   componentWillReceiveProps(nextProps) {
     const erroredUpdates = nextProps.refresh.statuses.failed;
     if (erroredUpdates.length === 0) {
-      this.props.submitForm(JSON.parse(sessionStorage.getItem('hr-form')));
-    }
-    if (nextProps.form.ready) {
-      this.props.router.push('/download');
+      this.submitAndDownload();
     }
   }
 
   componentWillUnmount() {
     clearInterval(this.pollRefresh);
+    clearTimeout(this.refreshTimeout);
   }
 
-  handleSkipToDownload(e) {
-    e.preventDefault();
+  submitAndDownload() {
```

Comment:
Should be fine in this case, but noticed this didn't get bound in the constructor. Also, you needed to do it in the `setTimeout` anyway.